### ISSUE TITLE
6. BERT Paper Issue

### ISSUE THREAD

**Issue by InhyeokYoo** [InhyeokYoo/NLP#2]
# 0. Why does BERT use Encoder ONLY architecture?
## Why don't just use both of them?

Inputs for both encoder and decoder are the same when you train for LM objective with an unsupervised dataset.  Therefore, the encoder and the decoder of the transformer will train redundant information from the dataset. See [this](https://github.com/openai/gpt-2/issues/157) for more details.

## What are the differences between the effects of encoder and decoder architecture?
Encoders more like AE ((denoise) auto-encoder) approach while decoders more like AR (auto-regressive) approches.

# 1. Introduction
## Why does Q.A. task belong token-level tasks rather than paraphrasing (i.e. sentence-level task)?

Because the task of Q.A. is to predict whether each token is the start/end token of the answer or not.

# 3. BERT
## Why does not BERT always replace “masked” words with the actual [MASK] token?

The original paper says:
> Although this allows us to obtain a  bidirectional pre-trained model, a downside is that we are creating a mismatch between pre-training and fine-tuning, since the [MASK] token does not appear during fine-tuning. To mitigate this,  we do not always replace “masked” words with the actual [MASK] token.

If the machine trains a task reconstructing the original input for the [MASK] token, the machine will not predict anything in the fine-tuning step since the machine trains only modify the [MASK] tokens into **proper words** rather than reconstruct while sentence or understand the context. Therefore, we do not always replace “masked” words with the actual [MASK] token for understanding language itself (i.e., context).
